Modules/_testcapi/unicode.c

```diff
@@ -1429,6 +1429,25 @@ unicode_comparewithasciistring(PyObject *self, PyObject *args)
     return PyLong_FromLong(result);
 }
 
+/* Test PyUnicode_EqualToUTF8() */
+static PyObject *
+unicode_equaltoutf8(PyObject *self, PyObject *args)
+{
+    PyObject *left;
+    const char *right = NULL;
+    Py_ssize_t right_len;
+    int result;
+
+    if (!PyArg_ParseTuple(args, "O|y#", &left, &right, &right_len)) {
+        return NULL;
+    }
```

Comment:
Apparently, the function crash if right is NULL, so for now, it's safer to always require 2 arguments:

```suggestion
    PyObject *left;
    const char *right;
    Py_ssize_t right_len;
    int result;

    if (!PyArg_ParseTuple(args, "Oy#", &left, &right, &right_len)) {
        return NULL;
    }
```

Reply:
It is intentionally, so we can check it for NULL.

Reply:
In that case, please initialize right_len. Otherwise, it's uninitialized if the function is called with a single argument.

Reply:
But it is not used in any case.

Reply:
In any case I am going to change "|y#" to "z#". The argument will be mandatory and the variable always initialized, as you requested!